pkg/provider/gcp/secretmanager/client.go

```diff
@@ -378,6 +382,61 @@ func (c *Client) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretData
 	return []byte(val.String()), nil
 }
 
+func (c *Client) getSecretMetadata(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	secret, err := c.smClient.GetSecret(ctx, &secretmanagerpb.GetSecretRequest{
+		Name: fmt.Sprintf("projects/%s/secrets/%s", c.store.ProjectID, ref.Key),
+	})
+
+	err = parseError(err)
+	if err != nil {
+		return nil, fmt.Errorf(errClientGetSecretAccess, err)
+	}
+
+	const (
+		annotations = "annotations."
+		labels      = "labels."
+	)
+
+	extractMetadataKey := func(s string, p string) string {
+		if !strings.HasPrefix(s, p) {
+			return ""
+		}
+		return strings.TrimPrefix(s, p)
+	}
+
+	if annotation := extractMetadataKey(ref.Property, annotations); annotation != "" {
+		v, ok := secret.GetAnnotations()[annotation]
+		if !ok {
+			return nil, fmt.Errorf("annotation with key %s does not exist in secret %s", annotation, ref.Key)
+		}
+
+		return []byte(v), nil
+	}
+
+	if label := extractMetadataKey(ref.Property, labels); label != "" {
+		v, ok := secret.GetLabels()[label]
+		if !ok {
+			return nil, fmt.Errorf("label with key %s does not exist in secret %s", label, ref.Key)
+		}
+
+		return []byte(v), nil
+	}
+
+	if ref.Property != "" {
+		return nil, fmt.Errorf("invalid property %s: metadata property should start with either %s or %s", ref.Property, annotations, labels)
```

Comment:
I would like to support the following case that allows an user to extract all key/value pairs from labels/annotations. This is currently not possible because the property doesn't match `labels.` or `annotations.` and hence returns this error.

```yaml
  dataFrom:
  - extract:
      metadataPolicy: Fetch
      key: dev-secret-test
      property: labels
```

Reply:
@moolen Thank you for your review and sorry that took me so long to fix the problem (I could not secure enough time this week to work on this issue 😮‍💨 ). I've fixed the PR again so would you review it when you have time?